src/main/java/jbash/filesystem/FileSystemObject.java

```diff
@@ -1,11 +1,44 @@
 package jbash.filesystem;
 
-import java.nio.file.FileSystem;
-
 public abstract class FileSystemObject {
```

Comment:
This should be package-private, since we don't want anyone else using this class: I'd change it to `abstract class FileSystemObject` (just remove the `public`.)

Reply:
where did that come from!